jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/Configuration.java

```diff
@@ -0,0 +1,221 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.websocket.core;
+
+import java.time.Duration;
+
+public interface Configuration
+{
+    /**
+     * Get the Idle Timeout
+     *
+     * @return the idle timeout
+     */
+    Duration getIdleTimeout();
+
+    /**
+     * Get the Write Timeout
+     *
+     * @return the write timeout
+     */
+    Duration getWriteTimeout();
+
+    /**
+     * Set the Idle Timeout.
+     *
+     * @param timeout the timeout duration (timeout &lt;= 0 implies an infinite timeout)
+     */
+    void setIdleTimeout(Duration timeout);
+
+    /**
+     * Set the Write Timeout.
+     *
+     * @param timeout the timeout duration (timeout &lt;= 0 implies an infinite timeout)
+     */
+    void setWriteTimeout(Duration timeout);
+
+    boolean isAutoFragment();
+
+    void setAutoFragment(boolean autoFragment);
+
+    long getMaxFrameSize();
+
+    void setMaxFrameSize(long maxFrameSize);
+
+    int getOutputBufferSize();
+
+    void setOutputBufferSize(int outputBufferSize);
+
+    int getInputBufferSize();
+
+    void setInputBufferSize(int inputBufferSize);
+
+    long getMaxBinaryMessageSize();
+
+    void setMaxBinaryMessageSize(long maxSize);
+
+    long getMaxTextMessageSize();
+
+    void setMaxTextMessageSize(long maxSize);
+
+    interface Customizer
+    {
+        void customize(Configuration configurable);
+    }
+
+    class ConfigurationHolder implements Configuration
+    {
+        protected Duration idleTimeout;
+        protected Duration writeTimeout;
+        protected Boolean autoFragment;
+        protected Long maxFrameSize;
+        protected Integer outputBufferSize;
+        protected Integer inputBufferSize;
+        protected Long maxBinaryMessageSize;
+        protected Long maxTextMessageSize;
```

Comment:
Yuck!

`protected` fields, getter and setters with `Integer` and `Long` rather than their primitive counterparts, etc.

I'm sure we can do better.

Reply:
The intention here is to tri-state the fields, because we have lots of sources of configuration - filters, individual mappings, endpoints, annotations, connection factories and ultimately defaults.     These configurations have to be merged into a single configuration, so knowing if they have been set is important when doing the merge - else we end up with many "defaults" distributed through out the code.